interview-questions.md

```diff
@@ -6,64 +6,72 @@ First, without external resources. Challenge yourself to answer from memory as i
 
-  Researched answer:
+  Researched answer: What I've collected from the internet is that map will go through each element, depending on the conditions of the code and return an altered data while preserving the length. Filter on the other hand will return the same or less amount of elements in array. What it will not do is change the value of the element. In my experience with coding, filter is the better alternative when the expected outcome is choosing a select few from the origin array. Recap, map will modify the elements while returning the same length, filter will choose and pick what to return. In the end, changing the length.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is a set of code that you give the computer to execute while method is where there already built in functionality. Both are very important when it comes to coding.
 
-  Researched answer:
+  Researched answer: I feel like both function and method are too similar for me to properly define each and identify the difference. What I do know for certain, is that functions are a block of code that coders write for the computer to execute.  I know that the function gets executed when it is invoked. Once the condition is met within the code block the value is returned. Methods are what can be done on objects. I would say that the main difference from my understanding is that the function can stand on its own while method is attached to an object.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Unfortunately, I get the general idea of object destructuring. It's when you take an object and break it down so that when you call on it you could call on the variable. You create a way where it is easily accessible rather then using dot notation.
 
-  Researched answer:
+  Researched answer: Object destructuring is when you take properties from objects and set them to variables. It's useful when you want to avoid the tedious act of writing out property(ies) with the object and dot notation. When you use object destructuring then you can get pass the extra coding.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I actually have no idea what hoisting is and I'm willing to find out.
 
-  Researched answer:
+  Researched answer: JavaScript Hoisting is when you bring variable, class, and declaration up to the very top of the scope. Hoisting does not work with initialization. From what I'm reading I feel like everyone in my cohort are practicing JavaScript Hoisting without being aware of it. It's a surprise to me that when I read that question, I knew nothing of it. Going through some websites and reading about it I can say that I use it. It's something that you definetely get a feel for the longer you code.
```

Comment:
It's also something to be very aware of as it can introduce bugs into code as well.